### DISCUSSION TITLE
TypeScript with stored procedures

### DISCUSSION THREAD

**Discussion by Nick-Mazuk** [supabase/supabase#1709] [Questions] [answered]
Short question today: is it currently possible to use TypeScript with stored procedures?

Something along the lines of…

```ts
supabase.rpc<definitions['procedure']>('procedure', { ...options })
```

I couldn't see anything in the docs or the generated type definitions for this.

**Comment by soedirgo** [comment, marked as answer]
I don't think this is possible right now @Nick-Mazuk, and supporting this also carries some cons. See e.g. https://github.com/jawj/zapatos/issues/58 for a discussion on generating TS types for functions/stored procedures.
